akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/PersistentActorContextImpl.scala

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright (C) 2017-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.typed
+package internal
+
+import java.time.Duration
+import java.util
+import java.util.{ Optional, function }
+import java.util.function.BiFunction
+
+import akka.actor.Cancellable
+import akka.actor.typed.{ ActorRef, ActorSystem, Behavior, Logger, Props }
+import akka.actor.typed.scaladsl.ActorContext
+import akka.annotation.InternalApi
+import akka.util.Timeout
+
+import scala.concurrent.ExecutionContextExecutor
+import scala.concurrent.duration.FiniteDuration
+import scala.reflect.ClassTag
+import scala.util.Try
+
+/**
+ * INTERNAL API
+ */
+@InternalApi private[akka] case class PersistentActorContextImpl[T](ac: ActorContext[T], lastSequenceNr: Long) extends PersistentActorContext[T] with javadsl.PersistentActorContext[T] with scaladsl.PersistentActorContext[T] with ActorContext[T] with akka.actor.typed.javadsl.ActorContext[T] {
+
+  def getLastSequenceNr: Long = lastSequenceNr
+
+  def asJavaPAC: javadsl.PersistentActorContext[T] = this
+
+  def asScalaPAC: scaladsl.PersistentActorContext[T] = this
```

Comment:
simply asScala / asJava?

Reply:
Won't work. asScala / asJava are already defined in the ActorContext.

Reply:
Yes, so since PersistentActorContextImpl extends ActorContext you can make the return type more specific than in the super class -- asScala should work, have you tried?

Reply:
Yes, I tried, but I think I did something wrong. This time I managed to do it. Thanks for catching this.

Reply:
Great, thanks!